components/authz-service/engine/opa/opa.go

```diff
@@ -621,19 +621,45 @@ func (s *State) SetPolicies(ctx context.Context, policies map[string]interface{}
 	return s.initPartialResult(ctx)
 }
 
-// Spike TODO: Can we have a separate method for just setting the rule mappings?
-// OR does the entire OPA store have to be re-evaluated at once. IF that's true,
-// should we have the same OPA instance in general for rules?
+// Q: Can we have a separate method for just setting the rule mappings?
+//    OR does the entire OPA store have to be re-evaluated at once. IF that's
+//    true, should we have the same OPA instance in general for rules?
+// A: This depends on whether it's used: If we imagine the rules-related queries
+//    to eventually check policies and their projects, it needs to be in the
+//    same store.
 //
-// V2SetPolicies replaces OPA's data with a new set of policies and roles,
-// and resets the partial evaluation cache for v2
+// V2SetPolicies replaces OPA's data with a new set of policies, rules, and
+// roles, and resets the partial evaluation cache for v2
 func (s *State) V2SetPolicies(
 	ctx context.Context, policyMap map[string]interface{},
 	roleMap map[string]interface{}, ruleMap map[string][]interface{}) error {
+
+	// here, we flatten the policy structure: let's see if it buys us something in
+	// opa performance
```

Comment:
Can we make an actual claim here yet (instead of a question) or do you want to see how it performs in the wild?

Reply:
Sorry that's what my "retract this benchmark" in the PR message and the `DO-NOT-MERGE` meant to convey (together with some hard-to-find slack backscroll): It doesn't help. I've opened an issue with OPA, trying to get further eyes on it. (https://github.com/open-policy-agent/opa/issues/1447)

Reply:
Sorry that's what my "retract this benchmark" in the PR message and the `DO-NOT-MERGE` meant to convey (together with some hard-to-find slack backscroll): It doesn't help. I've opened an issue with OPA, trying to get further eyes on it. (https://github.com/open-policy-agent/opa/issues/1447)